src/deluge/gui/views/instrument_clip_view.cpp

```diff
@@ -1282,93 +1282,112 @@ const uint32_t auditionPadActionUIModes[] = {UI_MODE_AUDITIONING,
 
 ActionResult InstrumentClipView::padAction(int32_t x, int32_t y, int32_t velocity) {
 
-	if (x == 15 && y == 2 && velocity > 0
+if (x == 15 && y == 2 && velocity > 0
 	    && runtimeFeatureSettings.get(RuntimeFeatureSettingType::DrumRandomizer) == RuntimeFeatureStateToggle::On) {
 		int32_t numRandomized = 0;
-		for (int32_t i = 0; i < 8; i++) {
-			if (getCurrentUI() == this && this->auditionPadIsPressed[i]) {
-				if (getCurrentInstrumentType() != InstrumentType::KIT) {
-					continue;
-				}
-				AudioEngine::stopAnyPreviewing();
-				Drum* drum = getCurrentInstrumentClip()->getNoteRowOnScreen(i, currentSong)->drum;
-				if (!drum || drum->type != DrumType::SOUND) {
-					continue;
-				}
-				SoundDrum* soundDrum = (SoundDrum*)drum;
-				MultiRange* r = soundDrum->sources[0].getRange(0);
-				if (r == NULL || ((MultisampleRange*)r)->sampleHolder.audioFile == NULL) {
-					continue;
-				}
-				AudioFileHolder* afh = r->getAudioFileHolder();
-				if (afh == NULL) {
-					continue;
-				}
+		if (sdRoutineLock) {
+			return ActionResult::REMIND_ME_OUTSIDE_CARD_ROUTINE;
+		}
+		if (currentSong->currentClip->output->type != InstrumentType::KIT) {
```

Comment:
Can use `getCurrentInstrumentType()` here.

Reply:
OK. Does that have further advantages apart from avoiding the train wreck style of programming?
But I agree that the optical improvement would be already be worth the change...

Reply:
Style is probably the primary motivator, but it also compiles to slightly smaller code since a function call is a single instruction (plus maybe one register transfer to set up the argument) versus several chained load instructions.

Reply:
as a heads up, I just renamed the recommend function to `getCurrentOutputType` (and the enum is `OutputType` rather than `InstrumentType`)